actionpack/lib/abstract_controller/base.rb

```diff
@@ -10,6 +10,15 @@ module AbstractController
   class ActionNotFound < StandardError
   end
 
+  # Raised when a nested respond_to is triggered.
+  class DoubleRespondToError < StandardError
```

Comment:
Should we raise on any nested double `respond_to` call? I'm not familiar with a good reason to do this. We could also rename this to `RespondToMismatchError ` if we are ok with allowing nested `respond_to`s with matching types.

Reply:
Agreed..I think I picked the name before I considered adding support for "compatible" nested `respond_to` calls. 

Speaking of which..do you think this is the right call? I did it to minimize "unnecessary breakage". But, the scenarios where nested calls are "compatible: is largely total coincidence. It isn't that one is coded correctly and one incorrectly. In general I'd bet that it is a code smell in most situations.